server/src/main/java/org/elasticsearch/TransportVersions.java

```diff
@@ -141,6 +141,7 @@ static TransportVersion def(int id) {
     public static final TransportVersion V_8_500_074 = def(8_500_074);
     public static final TransportVersion NODE_INFO_INDEX_VERSION_ADDED = def(8_500_075);
     public static final TransportVersion FIRST_NEW_ID_LAYOUT = def(8_501_00_0);
+    public static final TransportVersion RECOVERY_COMMIT_TOO_NEW_EXCEPTION_ADDED = def(8_501_00_1);
```

Comment:
@thecoop it seems we are now partially using a different format for TVs. Since this seems to be a first, could you double check if this is correct? I've just bumped the patch version.

Reply:
We are using a different format - see #99640. As in the comment above, the next version should be `8_502_00_0`

Reply:
Did I do an incorrect version bump in https://github.com/arteam/elasticsearch/commit/2998547c1d846ef99e1b7b02854cd42b34220071? Should I revert the commit? I haven't merged the PR that is supposed to depend on it.

Reply:
I see. Thanks.

Reply:
@thecoop it seems the same mistake has been made and it is on main already! See https://github.com/elastic/elasticsearch/blob/68ddb19c01a01e7ece03ef6cf52a8ef6c80a6270/server/src/main/java/org/elasticsearch/TransportVersions.java#L144

Reply:
so the very last part (`_P`) is for a patch release? IOW, in a `M_NXX_YY_P`, M is major, that's clear. Is P patch version as in 8.10.**2** ? and is N some sort of minor version? It seems for "normal" dev-related bumping (due to incompatible changes e.g.) we either increment XX or YY depending on stateful or serverless. Is that correct? M, N and P change due to releases?

Reply:
P is a patch to the transport protocol (for emergency fixes etc)
NNN is the version for base elasticsearch
YY is the version for serverless only